conans/test/integration/package_id/test_cache_compatibles.py

```diff
@@ -236,3 +236,50 @@ class Pkg(ConanFile):
               f"-s compiler.runtime={runtime}")
         assert "mylib/1.0: Main binary package 'e340edd75790e7156c595edebd3d98b10a2e091e' missing."\
                f"Using compatible package '{package_id1}'"
+
+    def test_fail_with_options_deleted(self):
+        c = TestClient(servers={"default": TestServer()}, inputs=["admin", "password"])
+        conanfile = textwrap.dedent("""
+            from conan import ConanFile
+            class Pkg(ConanFile):
+                name = "mylib"
+                version = "1.0"
+                package_type = "library"
+                options = {"shared": [True, False], "with_fmt_alias": [True, False], "header_only": [True, False]}
+                default_options = {"header_only": False, "shared": False, "with_fmt_alias": False}
+                settings = "os", "arch", "compiler", "build_type"
+                def package_id(self):
+                    if self.options.header_only:
+                        self.info.header_only()
+                    else:
+                        del self.info.options.with_fmt_alias
+                def package_info(self):
+                    print(self.options.with_fmt_alias)
+            """)
+        c.save({"conanfile.py": conanfile})
+        c.run("create . -s compiler.cppstd=14")
+        c.run("upload '*' -c -r default")
+        c.run("remove '*' -f")
+
+        consumer = textwrap.dedent("""
+            from conan import ConanFile
+            from conan.tools.build import check_min_cppstd
+            from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+            class consumer(ConanFile):
+                name = "consumer"
+                version = "1.0"
+                settings = "os", "compiler", "build_type", "arch"
+                options = {"shared": [True, False], "fPIC": [True, False]}
+                default_options = {"shared": False, "fPIC": True}
+                def requirements(self):
+                    self.requires("mylib/1.0")
+                """)
+        c.save({"conanfile.txt": consumer})
```

Comment:
This is most likely a bug in the test. You are saving a ``conanfile.txt``, that is being ignored, as ``conan create`` only reads ``conanfile.py``. 

I am cleaning this a bit

Reply:
Yes, sorry, that was the result of changing it back an forth and half of the test was useless, thanks for cleaning it.